### PR TITLE
Move participant to a different school

### PR DESCRIPTION
**Pull request by tonyheadford**
## Ticket and context

Helper service to move a participant to a new school, as discussed in dev retro, these are coming up often in support so this can make it simpler to handle from the console.

## Tech review
Please check this is doing the right thing especially regarding the sparsity and pupil premium uplift

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (lead-providers/guidance/release-notes)

### CODE REVIEW

**Review by jamesrose — APPROVED** [review]
💯

**Review by kicferk1 — APPROVED** [review]

File: app/services/move_participant.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def call
+    cohort = participant_profile.school_cohort.cohort
+    school_cohort = school.school_cohorts.find_by!(cohort: cohort)
+    return if school_cohort.blank?
+
+    ActiveRecord::Base.transaction do
+      participant_profile.teacher_profile.update!(school: school)
+      participant_profile.update!(school_cohort: school_cohort,
+                                  sparsity_uplift: school.sparsity_uplift?(cohort.start_year),
+                                  pupil_premium_uplift: school.pupil_premium_uplift?(cohort.start_year))
```
Comment (ethanmills) [inline app/services/move_participant.rb:20]: @blackrat do we want this?

Comment (blackrat) [reply app/services/move_participant.rb:20]: If it's to fix an error, yes. For people changing schools normally, no, they should have the original school values. The uplift payment is a "reward" for recruitment. If they change schools, that's a later thing that doesn't effect it.

Comment (blackrat) [reply app/services/move_participant.rb:20]: If it's an error, yes. If it's a genuine teacher changing schools, then no. It's a business rule based on where they were when they were recruited as opposed to where they actually are now.

**Pull request merged**